packages/teams-js/src/public/dialog.ts

```diff
@@ -98,6 +98,19 @@ export namespace dialog {
     sendMessageToParent('messageForParent', [message]);
   }
 
+  /**
```

Comment:
Is this being added just for testability? If so, it might be worth talking about. In general I don't like adding exported public functions that aren't actually needed for app developers because it makes the sdk surface harder to understand (this is the same as the PostMessageChannel that open returns if I'm not mistaken)

Reply:
This used to be for testability only, but I think we want to return a `Promise<void>` again for `dialog.open` and so we would need to extract this capability out.

Reply:
Are you planning to make those changes in a separate PR? (Turning the return to Promise<void> and resolving the promise when dialog registers to hear from app)?

Reply:
Promisifying will be in a separate PR, this PR enables us to test any changes and ensure we aren't breaking anything in the process.